src/Components/Components/src/PublicAPI.Unshipped.txt

```diff
@@ -1,10 +1,21 @@
 #nullable enable
+Microsoft.AspNetCore.Components.NavigationManager.AddLocationChangingHandler(System.Func<Microsoft.AspNetCore.Components.Routing.LocationChangingContext!, System.Threading.Tasks.Task!>! locationChangingHandler) -> void
 Microsoft.AspNetCore.Components.NavigationManager.HistoryEntryState.get -> string?
 Microsoft.AspNetCore.Components.NavigationManager.HistoryEntryState.set -> void
+Microsoft.AspNetCore.Components.NavigationManager.NotifyLocationChanging(string! uri, bool isNavigationIntercepted, System.Action<Microsoft.AspNetCore.Components.Routing.LocationChangingResult!>! continueNavigation) -> void
+Microsoft.AspNetCore.Components.NavigationManager.RemoveLocationChangingHandler(System.Func<Microsoft.AspNetCore.Components.Routing.LocationChangingContext!, System.Threading.Tasks.Task!>! locationChangingHandler) -> void
 Microsoft.AspNetCore.Components.NavigationOptions.HistoryEntryState.get -> string?
 Microsoft.AspNetCore.Components.NavigationOptions.HistoryEntryState.init -> void
 Microsoft.AspNetCore.Components.Rendering.RenderTreeBuilder.AddContent(int sequence, Microsoft.AspNetCore.Components.MarkupString? markupContent) -> void
 Microsoft.AspNetCore.Components.Routing.LocationChangedEventArgs.HistoryEntryState.get -> string?
+Microsoft.AspNetCore.Components.Routing.LocationChangingContext
+Microsoft.AspNetCore.Components.Routing.LocationChangingContext.CancellationToken.get -> System.Threading.CancellationToken
+Microsoft.AspNetCore.Components.Routing.LocationChangingContext.IsNavigationIntercepted.get -> bool
+Microsoft.AspNetCore.Components.Routing.LocationChangingContext.PreventNavigation() -> void
+Microsoft.AspNetCore.Components.Routing.LocationChangingContext.TargetLocation.get -> string!
+Microsoft.AspNetCore.Components.Routing.LocationChangingResult
+Microsoft.AspNetCore.Components.Routing.LocationChangingResult.Canceled.get -> bool
+Microsoft.AspNetCore.Components.Routing.LocationChangingResult.Exception.get -> System.AggregateException?
```

Comment:
Based on what I read so far, we might be able to get rid of this by going async all the way through on NavigationManager.OnLocationChanging and just returning `bool` from the method to signal the navigation was cancelled.

Reply:
> we might be able to get rid of this by going async all the way through on NavigationManager.OnLocationChanging

This is the part I'm still unsure about, per my reasoning in [this comment](https://github.com/dotnet/aspnetcore/pull/42638#discussion_r920372967).